In [1]:
import pandas as pd
spain_df = pd.read_csv('../data/spain.csv')

'FT' Questions:
    - How many games ended 0-1?
    - How many games ended in a tie?
    - What games had the most goals scored? (can see 11-1, 12-1, and 10-2)
    - What are the goal totals for a given team?
    - What team has scored the most goals? Goals per win?
    - What are some ways to quantify home team advantage?
    - Is it different per team? and do you compare them against their away record or other teams?

How many games ended 0-1?

In [6]:
spain_df[spain_df['FT']== '0-1']

Date  Season                home          visitor   HT   FT  \
53     1929-05-09    1928         Real Madrid     FC Barcelona  0-0  0-1   
130    1930-01-26    1929           CE Europa       Real Union  0-1  0-1   
241    1931-03-01    1930    Racing Santander     FC Barcelona  0-1  0-1   
321    1932-02-14    1931           CD Alaves  Athletic Bilbao  0-1  0-1   
330    1932-02-28    1931           CD Alaves      Real Madrid  0-1  0-1   
...           ...     ...                 ...              ...  ...  ...   
26135  2022-04-19    2021          Real Betis         Elche CF  NaN  0-1   
26140  2022-04-21    2021  Espanyol Barcelona   Rayo Vallecano  NaN  0-1   
26143  2022-04-21    2021       Real Sociedad     FC Barcelona  NaN  0-1   
26144  2022-04-24    2021        FC Barcelona   Rayo Vallecano  NaN  0-1   
26189  2022-05-22    2021           CD Alaves         Cadiz CF  NaN  0-1   

       hgoal  vgoal  tier   round group notes  
53         0      1     1  league   NaN   NaN  
130        0      1     1  league   NaN   NaN  
241        0      1     1  league   NaN   NaN  
321        0      1     1  league   NaN   NaN  
330        0      1     1  league   NaN   NaN  
...      ...    ...   ...     ...   ...   ...  
26135      0      1     1  league   NaN   NaN  
26140      0      1     1  league   NaN   NaN  
26143      0      1     1  league   NaN   NaN  
26144      0      1     1  league   NaN   NaN  
26189      0      1     1  league   NaN   NaN  

[1498 rows x 12 columns]

In [7]:
spain_df[spain_df['FT']== '1-0']

Date  Season                home             visitor   HT   FT  \
24     1929-03-10    1928     Arenas de Getxo     Athletic Bilbao  1-0  1-0   
37     1929-04-07    1928        FC Barcelona  Espanyol Barcelona  0-0  1-0   
66     1929-05-30    1928        FC Barcelona       Real Sociedad  1-0  1-0   
97     1929-12-08    1929  Espanyol Barcelona     Atletico Madrid  0-0  1-0   
132    1930-01-26    1929       Real Sociedad  Espanyol Barcelona  0-0  1-0   
...           ...     ...                 ...                 ...  ...  ...   
26126  2022-04-16    2021           CD Alaves      Rayo Vallecano  NaN  1-0   
26164  2022-05-08    2021     Atletico Madrid         Real Madrid  NaN  1-0   
26166  2022-05-10    2021          Granada CF     Athletic Bilbao  NaN  1-0   
26180  2022-05-15    2021          Celta Vigo            Elche CF  NaN  1-0   
26193  2022-05-22    2021          Sevilla FC     Athletic Bilbao  NaN  1-0   

       hgoal  vgoal  tier   round group notes  
24         1      0     1  league   NaN   NaN  
37         1      0     1  league   NaN   NaN  
66         1      0     1  league   NaN   NaN  
97         1      0     1  league   NaN   NaN  
132        1      0     1  league   NaN   NaN  
...      ...    ...   ...     ...   ...   ...  
26126      1      0     1  league   NaN   NaN  
26164      1      0     1  league   NaN   NaN  
26166      1      0     1  league   NaN   NaN  
26180      1      0     1  league   NaN   NaN  
26193      1      0     1  league   NaN   NaN  

[3064 rows x 12 columns]

There are 1,498 games that the home team won 1-0.
There are 3,064 games where the away team won 0-1.
There are 4,562 games where a team won by one goal and shut-out the other team.

How many games ended in a tie?

In [8]:
spain_df[spain_df['hgoal'] == spain_df['vgoal']]

Date  Season             home             visitor   HT   FT  \
3      1929-02-10    1928    Real Sociedad     Athletic Bilbao  1-1  1-1   
12     1929-02-24    1928  Arenas de Getxo          Real Union  1-0  1-1   
18     1929-03-03    1928     FC Barcelona     Arenas de Getxo  1-2  2-2   
20     1929-03-10    1928        CE Europa          Real Union  2-2  3-3   
23     1929-03-10    1928    Real Sociedad  Espanyol Barcelona  0-0  1-1   
...           ...     ...              ...                 ...  ...  ...   
26177  2022-05-15    2021  Atletico Madrid          Sevilla FC  NaN  1-1   
26179  2022-05-15    2021         Cadiz CF         Real Madrid  NaN  1-1   
26181  2022-05-15    2021        Getafe CF        FC Barcelona  NaN  0-0   
26185  2022-05-20    2021      Real Madrid          Real Betis  NaN  0-0   
26190  2022-05-22    2021       Granada CF  Espanyol Barcelona  NaN  0-0   

       hgoal  vgoal  tier   round group notes  
3          1      1     1  league   NaN   NaN  
12         1      1     1  league   NaN   NaN  
18         2      2     1  league   NaN   NaN  
20         3      3     1  league   NaN   NaN  
23         1      1     1  league   NaN   NaN  
...      ...    ...   ...     ...   ...   ...  
26177      1      1     1  league   NaN   NaN  
26179      1      1     1  league   NaN   NaN  
26181      0      0     1  league   NaN   NaN  
26185      0      0     1  league   NaN   NaN  
26190      0      0     1  league   NaN   NaN  

[6275 rows x 12 columns]

6,275 games ended in a tie.

What games had the most goals scored? (can see 11-1, 12-1, and 10-2)

In [23]:
spain_df['FT'].value_counts().tail(40)

8-0     21
8-1     20
6-3     20
3-5     16
1-6     13
0-6     13
9-0     12
7-3     11
8-2     11
2-6     10
6-4      8
4-5      8
3-6      7
8-3      6
9-1      5
10-1     5
0-7      4
0-8      4
10-0     3
5-5      2
9-2      2
4-6      2
2-8      2
2-7      2
1-7      2
11-2     1
9-4      1
1-8      1
3-8      1
5-6      1
9-5      1
6-6      1
7-4      1
9-3      1
10-3     1
11-1     1
7-5      1
12-1     1
4-7      1
10-2     1
Name: FT, dtype: int64

12-1 Seems to be the highest scoring game. 

There have been 3 games where the score was 10-0.

In [17]:
spain_df[spain_df['FT']== '12-1']

Date  Season             home       visitor   HT    FT  hgoal  \
225  1931-02-08    1930  Athletic Bilbao  FC Barcelona  6-1  12-1     12   

     vgoal  tier   round group notes  
225      1     1  league   NaN   NaN

In [16]:
spain_df[spain_df['FT']== '10-0']

Date  Season             home      visitor   HT    FT  hgoal  \
1071  1941-09-28    1941       Sevilla FC  Real Oviedo  5-0  10-0     10   
1168  1942-01-11    1941  Athletic Bilbao   Celta Vigo  3-0  10-0     10   
2787  1950-11-19    1950  Athletic Bilbao    UE Lleida  3-0  10-0     10   

      vgoal  tier   round group notes  
1071      0     1  league   NaN   NaN  
1168      0     1  league   NaN   NaN  
2787      0     1  league   NaN   NaN

In [18]:
spain_df[spain_df['FT']== '11-2']

Date  Season         home   visitor   HT    FT  hgoal  vgoal  \
5027  1960-02-07    1959  Real Madrid  Elche CF  5-1  11-2     11      2   

      tier   round group notes  
5027     1  league   NaN   NaN

In [19]:
spain_df[spain_df['FT']== '11-1']

Date  Season        home       visitor   HT    FT  hgoal  vgoal  \
936  1940-09-29    1940  Sevilla FC  FC Barcelona  4-1  11-1     11      1   

     tier   round group notes  
936     1  league   NaN   NaN

In [20]:
spain_df[spain_df['FT']== '0-8']

Date  Season                 home       visitor   HT   FT  hgoal  \
4920   1959-10-25    1959        UD Las Palmas  FC Barcelona  0-6  0-8      0   
21746  2010-11-20    2010           UD Almeria  FC Barcelona  0-5  0-8      0   
23496  2015-05-02    2014           Cordoba CF  FC Barcelona  NaN  0-8      0   
23867  2016-04-20    2015  Deportivo La Coruna  FC Barcelona  NaN  0-8      0   

       vgoal  tier   round group notes  
4920       8     1  league   NaN   NaN  
21746      8     1  league   NaN   NaN  
23496      8     1  league   NaN   NaN  
23867      8     1  league   NaN   NaN

In [21]:
spain_df[spain_df['FT']== '0-7']

Date  Season                 home       visitor   HT   FT  hgoal  \
4054   1956-02-05    1955  Deportivo La Coruna  FC Barcelona  0-2  0-7      0   
4407   1957-09-29    1957        UD Las Palmas  FC Barcelona  0-4  0-7      0   
5743   1963-02-10    1962            CD Malaga  FC Barcelona  0-2  0-7      0   
22364  2012-04-29    2011       Rayo Vallecano  FC Barcelona  0-3  0-7      0   

       vgoal  tier   round group notes  
4054       7     1  league   NaN   NaN  
4407       7     1  league   NaN   NaN  
5743       7     1  league   NaN   NaN  
22364      7     1  league   NaN   NaN

In [24]:
spain_df[spain_df['FT']== '9-0']

Date  Season              home             visitor   HT   FT  \
6      1929-02-17    1928   Athletic Bilbao  Espanyol Barcelona  3-0  9-0   
408    1933-01-29    1932  Racing Santander           CD Alaves  5-0  9-0   
516    1934-02-04    1933   Athletic Bilbao     Arenas de Getxo  3-0  9-0   
2608   1949-12-04    1949        Sevilla FC           Gimnastic  6-0  9-0   
2715   1950-09-17    1950         CD Malaga           UE Lleida  5-0  9-0   
3121   1952-02-10    1951      FC Barcelona      Sporting Gijon  4-0  9-0   
3629   1954-03-28    1953      FC Barcelona         Real Oviedo  7-0  9-0   
3911   1955-09-11    1955   Atletico Madrid         Hercules CF  3-0  9-0   
4427   1957-10-20    1957   Atletico Madrid       UD Las Palmas  4-0  9-0   
4745   1958-12-28    1958   Athletic Bilbao      Sporting Gijon  6-0  9-0   
4749   1959-01-04    1958   Athletic Bilbao          Celta Vigo  5-0  9-0   
10024  1979-01-14    1978      FC Barcelona      Rayo Vallecano  4-0  9-0   

       hgoal  vgoal  tier   round group notes  
6          9      0     1  league   NaN   NaN  
408        9      0     1  league   NaN   NaN  
516        9      0     1  league   NaN   NaN  
2608       9      0     1  league   NaN   NaN  
2715       9      0     1  league   NaN   NaN  
3121       9      0     1  league   NaN   NaN  
3629       9      0     1  league   NaN   NaN  
3911       9      0     1  league   NaN   NaN  
4427       9      0     1  league   NaN   NaN  
4745       9      0     1  league   NaN   NaN  
4749       9      0     1  league   NaN   NaN  
10024      9      0     1  league   NaN   NaN